scrapy_project/spiders/book_spider.py

```diff
@@ -0,0 +1,59 @@
+import re
+
+import scrapy
+from scrapy.http import Response
+
+
+class BookSpiderSpider(scrapy.Spider):
+    name = "book_spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> scrapy.Request:
```

Comment:
The return type annotation for the `parse` method should be `Generator` or `Iterable` instead of `scrapy.Request`, as this method yields requests.